sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -775,6 +778,57 @@ def _get_sample_count(self, n_samples):
             return ceil(n_samples * self.subsample)
         return n_samples
 
+    def _get_color_dict_from_values(
+        self, value_list, categorical_or_continuous: str, cmap_name: str = "Spectral"
+    ):
+        color_dict = {}
+        if categorical_or_continuous == "categorical":
+            mapping = {
+                np.unique(value_list)[i]: f"C{i}"
+                for i in range(len(np.unique(value_list)))
+            }
+            color_dict["color_list"] = [
+                mapping[value_list[i]] for i in range(len(value_list))
+            ]
+            color_dict["categorical_mapping"] = mapping
+        elif categorical_or_continuous == "continuous":
+            cmap = mpl.cm.get_cmap(cmap_name)
+            norm = mpl.colors.Normalize(vmin=value_list.min(), vmax=value_list.max())
+            color_dict["color_list"] = [
+                mpl.colors.to_hex(cmap(norm(value))) for value in value_list
+            ]
+            color_dict["norm"] = norm
+            color_dict["cmap"] = cmap
+        return color_dict
+
+    def _check_if_colors_by_feature(self, ice_lines_kw, preds):
+        # case 1: color can be a single color for all lines given as a str for all lines
+        color_dict = {}
+        if isinstance(ice_lines_kw["color"], str):
+            color_dict["color_list"] = np.repeat(ice_lines_kw["color"], preds.shape[0])
+        # case 2: color can be given as a list values for which color is created
+        elif isinstance(ice_lines_kw["color"], (list, np.ndarray, pd.Series)):
+            value_list = np.array(ice_lines_kw["color"])
+            # case 2.1 Sequence of str -> categorical
+            if isinstance(value_list[0], str):
+                color_dict = self._get_color_dict_from_values(value_list, "categorical")
+                color_dict["type"] = "categorical"
+            # case 2.1 Sequence of floats/int -> continuous
+            elif isinstance(value_list[0], (float, int)):
+                color_dict = self._get_color_dict_from_values(value_list, "continuous")
+                color_dict["type"] = "continuous"
+            else:
+                raise ValueError(
+                    "Please only use string or numerical values for color selection"
+                )
+        else:
+            raise ValueError(
+                "Undefined type for color attribute. Please pass either"
+                "a str or a list of strings/numerical"
+            )
+        ice_lines_kw["color_dict"] = color_dict
+        return ice_lines_kw
```

Comment:
du erzeugst hier absichtlich seiteneffekte. es wäre klarer, wenn du einfach nur `color_dict` zurück gibst und beim aufrufen der funktion dann z.b. in `ice_lines_kw["color_dict"]` abspeicherst